### DISCUSSION TITLE
Can't understand socket connection

### DISCUSSION THREAD

**Discussion by 2jumper3** [WalletConnect/walletconnect-monorepo#3857] [Developer Support] [answered]
Hi everyone, and thank you so much for taking the time to read this post!

Currently, I'm in the process of integrating WalletConnect into our app and have encountered a few challenges (to be honest, quite a few). Here are the issues I'm facing:

1. According to the documentation, a socket connection is required, but I'm unsure which server to use. I'm attempting to use `wss://relay.walletconnect.com`. Is this the correct one, or is there another server I should be using?

2. I've reviewed the documentation and the provided example in the SPM package, which seems straightforward:
```swift
import Starscream

extension WebSocket: WebSocketConnecting { }

struct SocketFactory: WebSocketFactory {
    func create(with url: URL) -> WebSocketConnecting {
        return WebSocket(url: url)
    }
}
```
However, in the actual project, it's generating various errors and requires many properties to be provided. Should I fill in these properties, or is something going wrong?
<img width="1028" alt="Screenshot 2023-11-01 at 20 48 07" src="https://github.com/WalletConnect/walletconnect-monorepo/assets/45697468/68b8c669-e2ae-4298-98be-ffde70c99dd3">

3. 
**UPDATED:**
I've wrote separate WebSocket layer, and got error:
`Error 1011` on `wss://relay.walletconnect.com/` which address I should use?)

Any guidance on the above issues would be greatly appreciated!
I use docs: 
https://docs.walletconnect.com/api/core/relay
https://medium.com/walletconnect/beginner-guide-to-walletconnect-v2-0-for-swift-developers-4534b0975218
And example:
https://github.com/WalletConnect/WalletConnectSwiftV2/tree/develop/Example

**Comment by alexander-lsvk** [comment, marked as answer]
@2jumper3 
1. Yes, the server is correct
2. Try to use Starscream 3.1.2

**Reply by 2jumper3** [reply]
1. How to get JWT token or how to auth?) Because server response with "unauthorised" error 
2. Ok, great, I'll try! 
Thanks, Alex!
